vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -1551,7 +1563,33 @@ public DefaultMutableTreeNode getTreeNode(Configurable target) {
   public void mouseDragged(MouseEvent evt) {
   }
 
+  /** Is the Component allowed to be moved? */
+  public boolean isMoveAllowed(Configurable target) {
+    return target.isMovable();
+  }
+
+  /**
+   * Is the Component allowed to be deleted?
+   * A mandatory component can be deleted if there is more than one of them with the same parent
+   * */
+  protected boolean isDeleteAllowed(Configurable target) {
+    if (target instanceof AbstractBuildable) {
```

Comment:
Could you simplify this by checking `target.isMandatory()` right at the start and returning `true` if not?

Reply:
I think this?
```
if (!target.isMandatory()) {
  return true;
}

if (target instanceof AbstractBuildable) {
  final Buildable parent = ((AbstractBuildable) target).getAncestor();
  if (parent instanceof AbstractBuildable) {
    final int count = ((AbstractBuildable) parent).getComponentsOf(target.getClass()).size();
    return count > 1;
  }
}

return false; 
```
Maybe that's not really simpler?

Reply:
Slightly more efficient though :) Change made.